src/webui/api/torrentscontroller.cpp

```diff
@@ -262,20 +262,48 @@ void TorrentsController::infoAction()
     for (const QString &hash : hashes)
         hashSet.insert(BitTorrent::InfoHash {hash});
 
+    const TorrentFilter torrentFilter(filter, (hashes.isEmpty() ? TorrentFilter::AnyHash : hashSet), category);
     QVariantList torrentList;
-    TorrentFilter torrentFilter(filter, (hashes.isEmpty() ? TorrentFilter::AnyHash : hashSet), category);
     for (BitTorrent::TorrentHandle *const torrent : asConst(BitTorrent::Session::instance()->torrents()))
     {
         if (torrentFilter.match(torrent))
             torrentList.append(serialize(*torrent));
     }
 
+    const auto lessThan = [](const QVariant &left, const QVariant &right) -> bool
+    {
+        Q_ASSERT(left.type() == right.type());
+
+        switch (static_cast<QMetaType::Type>(left.type()))
+        {
+        case QMetaType::Bool:
+            return left.value<bool>() < right.value<bool>();
+        case QMetaType::Double:
+            return left.value<double>() < right.value<double>();
+        case QMetaType::Float:
+            return left.value<float>() < right.value<float>();
+        case QMetaType::Int:
+            return left.value<int>() < right.value<int>();
+        case QMetaType::LongLong:
+            return left.value<qlonglong>() < right.value<qlonglong>();
+        case QMetaType::QString:
+            return left.value<QString>() < right.value<QString>();
+        case QMetaType::UnknownType:
+            return false;
+        default:
+            qWarning("Unhandled QVariant comparison, type: %d, name: %s", left.type()
```

Comment:
I would use Q_ASSERT_X instead. Also I would join default case with `QMetaType::UnknownType` (literally remove `QMetaType::UnknownType` case) since we don't expect to have it here as well.

Reply:
>I would use Q_ASSERT_X instead. 

I originally used that but I thought it would be hard to see what is wrong on a user's machine, so I opted for `qWarning` as it will emit messages to console (or system log). I mean I would expect users to report problems and (in this case) it is too hard to catch defects ourselves.

>(literally remove QMetaType::UnknownType case) since we don't expect to have it here as well.

But then the error message would be incorrect. IMO, unable to compare isn't equal to an unhanded case.
Or, for `UnknownType` case we can give it `Q_ASSERT_X(false, "unknown type error");`?

Reply:
About `UnknownType` etc.
There can be 2 kinds of unhandled types here:
1. We want to handle but just missed it (i.e. type of some existing field of serialized torrent);
2. We don't want to handle (i.e. no existing field of serialized torrent has this type).

We cannot easily and reliably distinguish these cases from within `lessThan` function itself. But we can and should cut off 2nd case before sorting (i.e. check for invalid value of `sort` parameter). And since `lessThan` function is local, we can only handle the first case inside it (i.e., assume that there is only one errored case possible due to missed type).

Reply:
> But we can and should cut off 2nd case before sorting (i.e. check for invalid value of `sort` parameter).

Validating `sort` parameter (`QString` type) is possible but it is very cumbersome. It would require a lot of string comparisons which is slow (and unnecessary).

>There can be 2 kinds of unhandled types here:

I think there are 3 kinds of them. You mentioned 2 and the other one is "unable to handle". When caught `UnknownType` what can we do? nothing, because it cannot be compared.
Also when `sort` parameter is some invalid value (due to typo or other defects), it will fall into this case and I don't think webAPI user will expect to see an unrelated error message. That is why I gave it `return false` and no error messages. 

>1. We want to handle but just missed it 

qbt emit some error/warning message so users can report it and we fix it.

>2. We don't want to handle 

We want to handle all types that are in use in our local case here. I think for our case here, the list would be empty.
And thankfully I don't need to handle list types (array, vector), otherwise I would put it in the "don't want to handle" list.

Reply:
> Validating `sort` & `sortedColumn` parameters (`QString` type) is possible but it is very cumbersome. It would require a lot of string comparisons which is slow (and unnecessary).

It doesn't look too cumbersome for me:
```c++
if (torrentList.size() == 0)
    return;
if (!sortedColumn.isEmpty())
{
    if (!torrentList[0].toMap().contains(sortColumn))
        throw;

    std::sort(/* So now only one erroneous case is possible here, i.e. if we miss to handle some type */);
}
```

Reply:
OK, PR updated.